pagopa/io-app
# [IOCOM-377] Asking notification permission in Android 13 and higher

## ⚠️ This pr depends on #4807 ⚠️
Please review first #4807 in order to avoid duplicate review work.

## Short description
This PR allows to **check** and **request** notification permissions on Android 13 and higher.

## List of changes proposed in this pull request
- Updated notification utils:
   - added function to check if permissions have been granted or not
   - added function to request permissions.
- Updated tests
- Fixed `ScrollView` regression of `OnboardingNotificationsPreferencesScreen` and minor improvements

## Preview

| new-user-accepted | new-user-refused |
|--------|--------|
| <video src="https://github.com/pagopa/io-app/assets/29163287/d506eeef-1ead-4548-b678-e1bbcea16558">  | <video src="https://github.com/pagopa/io-app/assets/29163287/8c1057ad-2b25-4681-bcea-0091481ccfaa">|

| old-user-accepted | old-user-refused |
|--------|--------|
| <video src="https://github.com/pagopa/io-app/assets/29163287/b3e51a36-f1d4-419d-b881-d63505037a1d">  | <video src="https://github.com/pagopa/io-app/assets/29163287/1f9fa7bc-d5f7-4931-bf2b-6e43c5c764b0">|

## How to test
Using the `io-dev-api-server`, set `firstOnboarding` to `true`/`false` and set `reminder_status` and `push_notifications_content_type` to `undefined`. During the onboarding flow, you should see the system permissions dialog for push notifications.
